src/main/groovy/net/vivin/gradle/versioning/tasks/TagTask.groovy

```diff
@@ -9,12 +9,23 @@ import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.TaskAction
 import org.gradle.tooling.BuildException
 
+import java.util.concurrent.Callable
+
 /**
  * Created on 6/23/16 at 10:26 AM
  * @author vivin
  */
 class TagTask extends DefaultTask {
     @Internal
+    Callable<?> message = { '' }
+
+    private Closure<String> fromEnvironmentVariableClosure = { System.env."$it" }
+
+    private Closure<String> fromSystemPropertyClosure = { System.properties."$it" }
+
+    private Closure<Object> fromProjectPropertyClosure = { project.hasProperty(it) ? project.property(it) : null }
```

Comment:
Not the old one, I meant this one. Ah, forget it, I confused the referenced one with the actual one here. #61 can be closed of course.